docs/scenarios_authoring/Intro.md

```diff
@@ -24,16 +24,16 @@ Every scenario starts with a source of data because we have to know what kind of
+In the **Streaming** [processing mode](/about/ProcessingModes) the data records processed by a scenario are called events. They are read from Kafka topics and processed by an [engine](/about/engines/) of choice: Flink or Lite. Events enter the scenario "via" a source node. The nodes process events; once the node finishes processing of an event, it hands it over to the next node in the processing flow. If there is a [split](./BasicNodes.md#split) node, the event gets "multiplied" and now two or more events "flow" in parallel through branches of the scenario.  There are also other nodes which can "produce" events; for example the [for-each](./BasicNodes.md#foreach) node or [time aggregate](AggregatesInTimeWindows.md) nodes. Finally, some nodes may terminate an event - for example the [filter](./BasicNodes.md#filter) node. The important takeaway here is that a single event that entered a scenario may result in zero, one or many events leaving the scenario (being written to Kafka topic).
 
-In the **Request-Response** processing mode it is the request data record which enters the scenario. The best and easiest way to understand how this request will be processed by the Nussknacker's scenario is to think of it as of a Streaming mode with just one event entering the scenario. All the considerations from the previous paragraph apply. The most important trait of a **Request-Response** scenario is that it's synchronous: some other computer system sends a request to Nussknacker and waits for a response. That request is the input to the scenario and the output - the decision - is the response. Since the other system is waiting for the response, there has to be exactly one. The natural question to ask is what happens if there are nodes in the scenario which "produce" additional data records - for-each or split. The topic of how to handle such situations is covered  [here](RRDataSourcesAndSinks.md#scenario-response-in-scenarios-with-split-and-for-each-nodes). 
+In the **Request-Response** processing mode it is a request data record which enters a scenario. The best and easiest way to understand how this request will be processed by Nussknacker's scenario is to think of it as of Streaming mode with a singular event. All the considerations from the previous paragraph apply. The most important trait of a **Request-Response** scenario is that it's synchronous: some other computer system sends a request to Nussknacker and awaits a response. That request is the input to the scenario and the output - the decision - is a response. Since the other system is awaiting a response, there has to be exactly one. The natural question to ask is what will happen when there are nodes in the scenario which "produce" additional data records - for-each or split. The topic of how to handle such situations is covered  [here](RRDataSourcesAndSinks.md#scenario-response-in-scenarios-with-split-and-for-each-nodes). 
```

Comment:
singular? What is this about? 

[https://www.askdifference.com/single-vs-singular/](https://www.askdifference.com/single-vs-singular/)

Reply:
I think singular is a better choice here as it outlines the fact that there's just one request processed. But maybe this wording is too complex for this kind of document.